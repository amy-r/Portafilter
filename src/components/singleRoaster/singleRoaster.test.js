/* eslint-disable */

import React from 'react';
import { SingleRoaster, mapStateToProps } from './SingleRoaster';
import { shallow } from 'enzyme'

describe('FormCard', () => {
  it('should exist', () => {
    const wrapper = shallow(
      <SingleRoaster 
        coffees={[{roaster: 'Corvus'}]}
        name='Corvus'
      />)

    expect(wrapper).toMatchSnapshot();
  })

  describe('MSTP', () => {
    it('should define the coffee props for the container', () => {
      const roasterName = "Corvus";
      const overAllScore= "10";
      const mockStore = {coffees: [ {roasterName, overAllScore} ]};
      const expected = [ {roasterName, overAllScore} ];
      const mapped = mapStateToProps(mockStore);
      expect(mapped.coffees).toEqual(expected)
    })
  })
})