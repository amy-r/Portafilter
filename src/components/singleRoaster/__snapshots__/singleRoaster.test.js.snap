// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCard should exist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    coffees={
        Array [
            Object {
              "roaster": "Corvus",
            },
          ]
    }
    name="Corvus"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           
          Corvus
           
</h1>,
        <div
          className="form-container"
>
          <Unknown
                    roaster="Corvus"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "Corvus",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Corvus",
          " ",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              roaster="Corvus"
/>,
          ],
          "className": "form-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "NaN",
            "nodeType": "function",
            "props": Object {
              "roaster": "Corvus",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             
            Corvus
             
</h1>,
          <div
            className="form-container"
>
            <Unknown
                        roaster="Corvus"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Corvus",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Corvus",
            " ",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                roaster="Corvus"
/>,
            ],
            "className": "form-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "NaN",
              "nodeType": "function",
              "props": Object {
                "roaster": "Corvus",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
